com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs

```diff
@@ -247,14 +257,43 @@ public ulong LocalClientId
 
         internal static event Action OnSingletonReady;
 
+
+
 #if UNITY_EDITOR
+        /// <summary>
+        /// Assures the ScenesInBuild asset exists and that it is always up to date
+        /// </summary>
+        /// <param name="isTesting"></param>
+        internal void PopulateScenesInBuild(bool isTesting = false)
+        {
+            // If we are testing or we are playing (in editor) and ScenesInBuild is null then we want to initialize and populate the ScenesInBuild asset.
+            // Otherwise, there are special edge case scenarios where we might want to repopulate this list
+            // The scenario with EditorApplication.isPlaying and ScenesInBuild being null is where we loaded a scene that did not have a NetworkManager but
+            // we transition to a scene with a NetworkManager while playing in the editor.  Under this condition we have to assign and populate.
+            if ( (!EditorApplication.isPlayingOrWillChangePlaymode && !EditorApplication.isPlaying && !EditorApplication.isUpdating) || isTesting
```

Comment:
Would like to see this `isTesting` go away.

My suggestion:

```c#
interface IScenesInBuildProvider
{
    bool ShouldBuildSceneList();
    ScenesInBuild InitializeScenesInBuild();
}

class ScenesInBuildProvider
{
    public bool ShouldBuildSceneList()
    {
        return (!EditorApplication.isPlayingOrWillChangePlaymode && !EditorApplication.isPlaying && !EditorApplication.isUpdating);
    }
    
    public ScenesInBuild InitializeScenesInBuild()
    {
        return ScenesInBuild.InitializeScenesInBuild();
    }
}

struct NetworkManagerOptions
{
    IScenesInBuildProvider ScenesInBuildProvider;
}

class NetworkManager
{
    private IScenesInBuildProvider m_ScenesInBuildProvider;
    
    NetworkManager(NetworkManagerOptions options = default)
    {
        m_ScenesInBuildProvider = options.ScenesInBuildProvider ?? new ScenesInBuildProvider();
    }
}
```

Then for testing, you would create a mock and initialize NetworkManager with it like so:

```c#

class TestScenesInBuildProvider : ScenesInBuildProvider
{
    public bool ShouldBuildSceneList()
    {
        return true;
    }
}

var NetworkManager = new NetworkManager(new NetworkManagerOptions { ScenesInBuildProvider = MockScenesInBuildProvider });
```

That way there's no custom logic in the shipped runtime that exists just for testing.